python/lib/modeldata/dmod/test/test_geopackage_hydrofabric_domain_detector.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+
+from dmod.core.meta_data import DataFormat, StandardDatasetIndex
+from dmod.core.dataset import ItemDataDomainDetector
+from ..modeldata.data.item_domain_detector import GeoPackageHydrofabricDomainDetector
+
+from . import find_git_root_dir
+
+
+class TestGeoPackageHydrofabricDomainDetector(unittest.TestCase):
+
+    def setUp(self):
+        self.detector_subclass = GeoPackageHydrofabricDomainDetector
+        self.expected_data_format = DataFormat.NGEN_GEOPACKAGE_HYDROFABRIC_V2
+        self.hyfab_ver = "2.0.1"
+
+        # Setup example 0
+        self.example_data = {0: find_git_root_dir().joinpath("data/example_hydrofabric_2/hydrofabric.gpkg")}
```

Comment:
This is unrunnable if this is used from a set of downloaded source and not by cloning.

Reply:
I'm not immediately 100% sure, but I highly doubt that is a supported use case (for the tests at least).  This may or may not be clearly documented, which would be a fair request.  

Given that something similar is done in several other spots already, I'm hesitant to take it out of this PR, as developing an alternative for what this fundamentally is supporting (dynamic detection of the paths of test data) is outside the scope.  And I'm pretty confident it would be worse to remove the dependent tests than to have tests that can't be run outside of standard development environments.  

That said, we can explore alternatives, but that needs to be tracked in its own issue.

Reply:
I agree that it wouldn't be wise to start ripping out unit tests - we should _not_ be adding more tests that rely upon a vcs though. I'd be far more comfortable if:

1. `find_git_root_dir` was renamed to something more general. This will divorce that name/purpose of the function from the implementation
2. Add a ticket to replace the innards of it to a VCS-free solution
3. Add a ticket to remove _all_ dependencies on `git` that are intertwined in any sort of general workflow (side scripts that are rarely run and don't feed into other scripts are probably fine)
4. Address the ticket in 2 asap

The purpose of `find_git_root_dir` is important - finding the test data is _kinda_ important when running tests. The implementation causes problems however and is already causing issues in other contexts.

Reply:
I don't have any issue with switching to a more general alternative, if one can be found.  I opened #570 to track exploring options.  But I have to disagree with the idea of needing to prohibit Git outright, so I don't think we should prioritize #570 over many of the other things in our queue.

DMOD development fundamentally relies on Github and Git; the idea that Git is unreasonable for things that are specific to DMOD development is paradoxical.  Maybe avoiding Git makes sense in certain functionality that's part of running deployments, but not for tests (if I've missed one of these functions using Git that isn't just for testing, let me know).

On top of that, since Docker is another dependency, there is no scenario where a DMOD deployment would not have access to Git.  It may not be the usual access, but we've already abandoned the usual setup if we downloaded the sources instead of cloning them.  There are even several custom DMOD Docker images that install and use Git.  So even in non-standard or atypical environments, it is feasible and relatively straightforward to adjust things so that these tests can be run as they are.

Reply:
>  But I have to disagree with the idea of needing to prohibit Git outright

I don't think we should prohibit `git` - it's just not appropriate for code that **must** run as part of the application. It being in some of the scripts used to coordinate changes wouldn't be a big deal.

> DMOD development fundamentally relies on Github and Git; the idea that Git is unreasonable for things that are specific to DMOD development is paradoxical. Maybe avoiding Git makes sense in certain functionality that's part of running deployments, but not for tests (if I've missed one of these functions using Git that isn't just for testing, let me know).

`git` commands are used in tests as of this PR (I don't remember seeing them in other tests, but I'm not omniscient over the codebase) and it's used as a fundamental step in some of the core scripts. If you're not in a `git` repo, for instance, you can't run `update_packages.sh`, making a plain jane playground environment from an archive impossible.

Also, **we** fundamentally rely on `git` and github. **We** are not the product. It's just a tool. Almost like a brand of a tool.  If we say that we fundamentally rely on github, as you've mentioned, that completely rules out GitLab and Gerrit, two tools that are completely reasonable in this context. There are also several other VCSs out there. Pledging allegiance to `git` (even though, I agree, 95% of devs/people poking around at it, will be coming in via a `git clone`) is like saying that one may _only_ use Black and Decker drills to put something together and that you cannot simply use a screwdriver. A screwdriver here is a pita, but it'd be kinda silly not to allow it if you're just poking around.

> There are even several custom DMOD Docker images that install and use Git

And that occurs within images where there is a finer level of control. Works great. It doesn't rely on the host here.

Calling `git` to get (ha) project paths works great - there's no denying it, but firmly coupling to it with no possible fallback is unreasonable. By all means, keep the `git` check, but there _has_ to be a fallback for when it's not there. 

We could also avoid **all** of this by just using pytest. Getting that is just a matter of calling `pip` - easy peasy.

Reply:
@christophertubbs, I feel like we can consider this conversation resolved, given separate discussions and issues #570 and #572, but I thought I should double check just to be safe.

Reply:
As long as there's attention paid to those tickets, rock on.